cli/cycpp.py

```diff
@@ -1305,8 +1352,34 @@ def impl(self, ind="  "):
             if 'derived_init' in info:
                 impl += ind + info['derived_init'] + '\n'
             else:
+                labels = info.get('alias', None)
+                if labels is None:
+                    labels = member if isinstance(t, STRING_TYPES) else [member]
                 reader = self.readers.get(t, self.readers.get(t[0], None))
-                impl += reader(member, alias, t, d, uitype, ind)
+
+                # generate condition to choose default vs given val if default exists
+                if d is not None:
+                    name = labels if isinstance(t, STRING_TYPES) else labels[0]
+                    impl += ind + 'if (sub->NMatches("{0}") > 0) {{\n'.format(name)
```

Comment:
So this part of the code - or its body - does not produce properly indented, human readable code.  I just opened up the generated file and I did not find it easy to read or understand.  From sink.h, here is what I am seeing:

```C++
  virtual void InfileToDb(cyclus::InfileTree* tree, cyclus::DbInit di) {
    cyclus::Facility::InfileToDb(tree, di);
    cyclus::InfileTree* sub = tree->SubTree("config/*");
    int i;
    int n;
    if (sub->NMatches("recipe_name") > 0) {
{      std::string recipe_name_val = cyclus::Query<std::string>(sub, "recipe_name");
recipe_name = recipe_name_val;}    } else {
      std::string recipe_name_tmp("");recipe_name = recipe_name_tmp;    }
{cyclus::InfileTree* bub = sub->SubTree("in_commods");
cyclus::InfileTree* sub = bub;
int n = sub->NMatches("val");
    std::vector< std::string > in_commods_val;
    in_commods_val.resize(n);
    for (i = 0; i < n; ++i) {
    std::string elem;{        std::string elem_in = cyclus::Query<std::string>(sub, "val", i);
elem = elem_in;}      in_commods_val[i] = elem;
    }
in_commods = in_commods_val;}{    double capacity_val = cyclus::Query<double>(sub, "capacity");
capacity = capacity_val;}    if (sub->NMatches("max_inv_size") > 0) {
{      double max_inv_size_val = cyclus::Query<double>(sub, "max_inv_size");
max_inv_size = max_inv_size_val;}    } else {
      double max_inv_size_tmp = 1e+299;max_inv_size = max_inv_size_tmp;    }
    di.NewDatum("Info")
    ->AddVal("recipe_name", recipe_name)
    ->AddVal("in_commods", in_commods)
    ->AddVal("capacity", capacity)
    ->AddVal("max_inv_size", max_inv_size)
    ->Record();
  };
```